style: apply /style-guide pass to models/runs (#2724)
wandb/docs: models/runs/delete-runs.mdx

description: Delete runs from a W&B project using the W&B App or the Public API, and learn how deleted run data is removed from storage.
---

> Delete runs from a W&B project interactively in the W&B App or programmatically with the W&B Public API. This page also explains when W&B removes deleted run data from object storage on self-managed deployments.

Reviewer: Public API is a subset of Python SDK. Also, second sentence sounded confusing.

```suggestion
This page shows how to delete runs from a W&B project interactively in the W&B App or programmatically with the W&B Python SDK. This page also explains the logic W&B uses to remove deleted run data from object storage on self-managed deployments.

Based on your use case, select one of the following tabs to learn how to delete a run:
```

Author: Applied — Public API to Python SDK, added the tab lead-in, and updated the frontmatter description to match.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

<Tabs>
<Tab title="W&B App" value="ui">
1. Navigate to the project that contains the runs you want to delete.
2. Select the **Runs** tab.
3. Select the checkbox next to the runs you want to delete.
4. Select the **Delete** button (trash can icon) above the table.
5. From the drawer that appears, select **Delete**.

> W&B removes the selected runs from the project.

Reviewer: ```suggestion
```

Author: Applied — removed the sentence.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

For projects that contain many runs, use the search bar to filter runs with Regex, or use the filter button to filter runs by status, tags, or other properties.
</Tab>

<Tab title="Python" value="python">

> You can delete runs programmatically with [`Run.delete()`](/models/ref/python/public-api/run#method-run-delete). To also remove artifacts associated with the run, set `delete_artifacts=True`.

Reviewer: ```suggestion
You can delete runs programmatically with [`wandb.apis.public.Run.delete()`](/models/ref/python/public-api/run#method-run-delete). To also remove artifacts associated with the run, set `delete_artifacts=True`.
```

Author: Applied — now `wandb.apis.public.Run.delete()`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

## When deleted run data is removed from storage

> When you delete a run, W&B removes it from the W&B App, but the underlying data can persist in object storage for a configurable retention period on self-managed deployments. This section describes how that retention works and what to expect for bucket usage after deletion.

Reviewer: Someone would need to verify this/approve the wording. Given the implications of it being wrong (since no one validating) , remove. 

Future: don't touch docs that revolve around data, security, and/or other sensitive info since in all likelihood they (the PRs that added said content) went under careful scrutiny from the Eng Team or Legal.

```suggestion
```

Author: Applied — removed the unvalidated data-retention summary sentence.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> On [W&B Dedicated Cloud](/platform/hosting/hosting-options/dedicated-cloud) and [W&B Self-Managed](/platform/hosting/hosting-options/self-managed), the `GORILLA_DATA_RETENTION_PERIOD` environment variable controls how long W&B retains deleted run data before permanently removing it from object storage. This setting doesn't remove artifacts; artifacts follow the deletion and garbage collection flow described in [Delete an artifact](/models/artifacts/delete-artifacts).

Reviewer: Don't use semicolons. It reduces readability (e.g. ESL folks)

Author: Fixed — split the semicolon clause into two sentences.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
